### DISCUSSION TITLE
조직 내 디자인 시스템에 대한 문제점 파악 및 진단

### DISCUSSION THREAD

**Discussion by peanudge** [InnerCircleA/design-system#9] [Show and tell]
**각자 조직 내 디자인 시스템에 대한 문제점 및 진단** 내용을 공유해주세요!🤚

저는 각자의 회사 상황도 함께 알려주시면 좋을 것 같아요.

**Comment by peanudge** [comment]
##  현재 회사 규모 및 개발 리소스 상황

현재 작성자의 회사는 소규모 스타트업이며 시리즈 A 투자를 받고 전체 인원 30~40명 중 Product 팀의 규모는 10명 내외 정도입니다. 그리고 크게 3개의 팀(2개의 사업팀, 1개의 플랫폼팀)으로 나눠져있으며 작성자는 하나의 사업팀에서 백엔드 개발자 1명, 디자이너 1명, 프론트엔드 개발자 1명으로 구성된 팀에서 프론트엔드 개발을 주력으로 담당하고 있습니다.

추가적인 상황으로 프론트엔드 개발자 두 명이 컨셉이 다른 리액트 네이티브 어플리케이션 각 1개씩 개발하고 있습니다. 

## 디자인 시스템 상황 및 진단

현재 회사에서는 개발자들은 디자인 시스템을 활용하고 있지않다고 고려되어 디자인부터 실제 구현부터 배포까지 프로세스 중 문제점을 확인해보겠습니다.

모든 디자이너들은 매주 함께 디자인 피드백을 해주는 시간을 갖고 있으며 기본적인 파운데이션과 여러 프로젝트에서 재사용 중인 컴포넌트가 피그마에 정리되있는 모습을 확인할 수 있습니다. 새로운 스프린트가 시작되면 디자이너들은 PM과  디자인 시안과 기획서(PRD)를 작성하고 검토하게 됩니다. (해당 단계에서 개발자도 참여하긴하지만 큰 역할은 하지않음)

그리고 실제 컴포넌트 구현 단계로 넘어갈 때 각 팀에 나눠져있는 엔지니어에게 피그마 링크를 공유하고 함께 미팅을 통해 피드백을 주고 받게됩니다. 이렇게 디자이너를 통해 전달받은 피그마 결과물(디자인 시안)을 **각 팀의 엔지니어가 다른 방식의 컴포넌트를 구현**을 진행하게됩니다. 

**Look And Feel** 을 위해 공통 라이브러리화를 시도했었던 흔적들이 보였지만 디자인 컨셉이 다른 두개의 product는 공통 라이브러리 번들 사이즈만 커지는 결과로 이어져 다시 각 프로젝트에서 컴포넌트를 관리하는 방향으로 이어졌습니다. 리액트 네이티브의 컴포넌트 라이브러리는 리액트 네이티브 외의 활용도가 크지도 않았고 디자인들의 성격이 많이 상이해서 재사용성의 이점이 없었다고 판단되었습니다.

그리고 디자이너는 과거에 디자인됐던 컴포넌트들이 어디 쓰이는지 몰라서 컴포넌트의 사용처를 엔지니어에게 구두로 물어봐야하는 상황이 발생해서 커뮤니케이션 비용이 매우 큰 상황입니다.  

결론,  
- 현재 디자인 시스템은 디자이너들만 하고 있고 엔지니어는 개별적으로 팀별로 진행하는 상황 - 일관성 저하 및 리소스 낭비
-  각 프로젝트의 디자인 성격이 너무 달라서 공통화가 쉽지않다. - 디자인 일관성 저하
- 새로운 디자이너가 기존 컴포넌트들의 사용처를 찾기가 힘들다. - 커뮤니케이션 비용 상승

**Reply by gyutae100** [reply]
네이트브 <- 오타 발견했어요!  

저는 10명 미만 소규모 회사만 다녀서 개발자로서 디자이너와 협업한 경험이 거의 전무해서, 디자인 시스템의 '시스템' 측면을 제대로 공감하기 힘들었는데,
지호님은 실제 경험으로 디자인 시스템의 필요에 대한 공감이 있으실거라고 보여요.

유익한 내용 공유 감사해요.

**Comment by Jeewhan** [comment]
##  현재 회사 규모 및 개발 리소스 상황

재직중인 회사는 시리즈 A 투자를 받았고, 전체 인원 30~40명 정도입니다. 제품과 직접적으로 연관된 조직은 R&D, Infra, 두 개의 제품팀(B2C Application, B2B SaaS)입니다. 작성자는 그 중 B2B 팀에서 제품책임자 1, 백엔드 2, 프론트엔드 2, 디자이너 2로 구성된 팀에서 프론트엔드 엔지니어를 맡고 있습니다.

## 디자인 시스템 상황 및 진단

현재 회사에는 디자인 시스템이 존재하지 않습니다. 지난 주에 읽은 '유난한 도전'에서 아래와 같은 문장이 있었습니다.

> 본격적으로 TDS를 구축하는 데에는 6개월 이상 소요됐다. 처음에는 앱에 깔린 수많은 파란색을 통일하는 데 주안점을 두고 시스템을 스케치하기도 했지만, 곧 방향을 틀었다. 사용자에게 일관성 있는 제품 경험을 선사하기 위해서는 밑그림부터 완전히 달라져야 했다. (...) 토스가 만드는 제품 하나하나가 종적인 혁신이라면, 각 제품이 일관성을 갖추고 최상의 사용자 경험을 제공하는 것은 횡적인 혁신이었다. (...) 정희연은 "돌이켜보면 이 시기는 디자인 시스템을 만들기에 이르지도 늦지도 않은 적당한 때였다"고 말했다. 제품이 너무 많아진 뒤였다면 시스템을 적용하는 데 너무 큰 비용과 시간이 들 터였다. 처음 TDS를 만들었을 때 이를 사용하는 디자이너는 7명이었고, 그 영향을 받는 MAU는 400만 명이었다. 3년여가 흐른 2021년 말에는 70명 넘는 디자이너가 TDS를 쓰고, 매달 1200만 명 넘는 사용자에게 영향을 주었다.

위에 나온 제품이 MAU 또는 리텐션의 상승세가 꺾여서 플랫폼 내 통일된 경험을 제공하는 것이 비즈니스적 목표가 된 시점을 맞이한 것에 비해, 만들고 있는 제품은 기능의 성숙도 또는 비즈니스 달성율이 초기 단계여서 요구사항의 변화가 굉장히 빈번하게 발생합니다. 따라서 당장 직면한 요구사항을 구현하는 것 외에 리소스를 투입하는 것은 팀에서 거의 허용되지 않습니다. 그나마 엔지니어들은 종종 여유가 발생하여 필요시 리팩터링 등도 진행할 수 있는 것에 비해, 디자이너의 리소스는 턱없이 부족한 상황입니다.

현재 디자이너 두 분은 제품의 Delivery 외에도 Discovery 에도 많은 리소스를 쓰고 있고, 필요에 따라선 다른 제품팀이나 비즈니스적인 필요에 의한 작업까지 진행하고 있는 상황입니다. 그래서 서로 Sync를 맞추기 위해 노력하고 계시지만 그럼에도 서로가 이전에 작업한 의도를 충분히 공유하지 못하고, 시안을 보고 추측하는 경우가 많아, 해당 페이지에 대한 수정이 이루어질 때 혼란이 종종 발생합니다. 또한 컴포넌트 기반으로 작업되고 있지 않아서 비슷한 시각적 요소이지만 (디자이너의 의도와 무관하게) 스타일이나 기능이 조금씩 다른 경우가 자주 발생합니다. 마지막으로 명확한 디자인 파운데이션이 색상 외에는 존재하지 않고, 색상조차도 종종 파운데이션에 없는 색상이 쓰이곤 합니다. 그래서 프로덕트 내 일관성이 부족한 상황입니다.

또한 디자인 시안이 Zeplin을 통해 전달되었을 때 상황에 따라 다른 프론트엔드 엔지니어가 작업을 담당하게 되어, 동일한 컴포넌트를 중복으로 개발하게 되기도 합니다. 또한 설령 동일한 시각적 또는 기능적 요소임을 알게 되더라도 디자이너와 합의하여 컴포넌트를 정의할 수 있는 상황이 못 되므로 섣불리 지금 파악된 요구사항만을 가지고 컴포넌트화 하는 것을 시도하는 것을 주저하게 됩니다.

마지막으로 디자이너가 놓친 디테일들을 엔지니어 단에서 수정하고 배포하다보니 디자인 시안과 실제 배포된 결과물이 다른 경우가 자주 발생합니다. 이러한 차이점에 대해 디자이너에게 공유는 하지만, 디자인 시안이 최신화되지 않을 때가 많고, 그것을 기반으로 또 다른 디자인 시안이 나오게 되어, 해당 작업에 관여했던 사람이 아닌 경우에는 디자인 시안이 맞는데 배포결과물이 잘못된 것인지, 디자인 시안이 최신화되지 않은 것인지에 대해 혼란을 겪게 됩니다.
